### PR TITLE
Added git detection handler

### PR DESCRIPTION
**Pull request by afrje**
#### Link to ticket

No ticket

#### Description

Added git detection handler

#### Screenshot of the result

If your change affects the user interface you should include a screenshot of the result with the pull request.

#### Checklist

- [ ] My code is covered by test cases.
- [ ] My code passes our test (all our tests).
- [ ] My code passes our static analysis suite.
- [ ] My code passes our continuous integration process.

If your code does not pass all the requirements on the checklist you have to add a comment explaining why this change 
should be exempt from the list.

#### Additional comments or questions

If you have any further comments or questions for the reviewer please add them here.

### CODE REVIEW

File: src/Controller/Admin/DashboardController.php [hunk 1]
```diff
@@ -51,6 +52,7 @@ public function configureMenuItems(): iterable
         yield MenuItem::linkToCrud('Domains', 'fas fa-link', Domain::class);
         yield MenuItem::linkToCrud('Servers', 'fas fa-server', Server::class);
         yield MenuItem::linkToCrud('Detection Results', 'fas fa-upload', DetectionResult::class);
+        yield MenuItem::linkToCrud('Git', 'fas fa-folder', Git::class);
```
Comment (turegjorup) [inline src/Controller/Admin/DashboardController.php]: Consider using one of the icons here: https://fontawesome.com/icons/github-square?s=brands

E.g. dp `'fas fa-github'`

File: src/Command/ReplayDetectionResultsCommand.php [hunk 2]
```diff
@@ -116,4 +136,13 @@ private function contextBuilder(): array
...
         ];
     }
+
+    protected function configure(): void
+    {
+        $this
+            ->addOption('type',
+                't',
+                InputOption::VALUE_REQUIRED,
+                'Limit replay to this type. Options are ['.join(', ', DetectionType::CHOICES).']');
```
Comment (turegjorup) [inline src/Command/ReplayDetectionResultsCommand.php]: Suggestion: It's also possible to use https://symfony.com/doc/current/components/console/helpers/questionhelper.html to give the user a "dropdown"

File: src/Controller/Admin/GitCrudController.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+use EasyCorp\Bundle\EasyAdminBundle\Controller\AbstractCrudController;
+use EasyCorp\Bundle\EasyAdminBundle\Field\AssociationField;
+use EasyCorp\Bundle\EasyAdminBundle\Field\DateTimeField;
+use EasyCorp\Bundle\EasyAdminBundle\Field\TextField;
+
+class GitCrudController extends AbstractCrudController
+{
+    public static function getEntityFqcn(): string
+    {
+        return Git::class;
```
Comment (turegjorup) [inline src/Controller/Admin/GitCrudController.php:22]: We should choose if this is `Git` or `Github`. I lean towrads `Github` but am open to arguments against

File: src/Controller/Admin/GitCrudController.php [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            ->remove(Crud::PAGE_DETAIL, Action::DELETE);
+    }
+
+    public function configureFields(string $pageName): iterable
+    {
+        yield AssociationField::new('server');
+        yield TextField::new('rootDir')->setColumns(6);
+        yield TextField::new('remote')->setColumns(6);
+        yield TextField::new('tag')->setColumns(6);
+        yield TextField::new('changes')->setColumns(6); // ->hideOnIndex();
```
Comment (turegjorup) [inline src/Controller/Admin/GitCrudController.php]: As you suggested. This should propably by hidden on the index page. 

You can do a virtual field to just highlight if there is changes.

File: src/Command/ReplayDetectionResultsCommand.php [hunk 5]
```diff
@@ -39,9 +41,27 @@ protected function execute(InputInterface $input, OutputInterface $output): int
     {
         $io = new SymfonyStyle($input, $output);
 
+        $detectionResults = $this->entityManager->createQueryBuilder()
```
Comment (turegjorup) [inline src/Command/ReplayDetectionResultsCommand.php]: Suggest `$queryBuilder`

File: src/Controller/Admin/GitCrudController.php [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        yield DateTimeField::new('createdAt');
+    }
+    /*
+    public function configureFields(string $pageName): iterable
```
Comment (turegjorup) [inline src/Controller/Admin/GitCrudController.php]: Remove please

**Review by turegjorup — CHANGES_REQUESTED** [review]

File: composer.json [hunk 7]
```diff
@@ -50,6 +50,7 @@
         "symfony/maker-bundle": "^1.37",
         "symfony/phpunit-bridge": "^6.1",
         "symfony/stopwatch": "^6.1",
+        "symfony/var-dumper": "6.1.*",
```
Comment (turegjorup) [inline composer.json]: Please use the `^6.1` format as above. This allows semver safe updates

File: composer.json [hunk 8]
```diff
@@ -100,9 +101,6 @@
         "post-update-cmd": [
             "@auto-scripts"
         ],
-        "apply-coding-standards": [
-            "vendor/bin/php-cs-fixer fix"
```
Comment (turegjorup) [inline composer.json:105]: Why is this removed?

File: src/Handler/GitHandler.php [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+                    'url' => $remote,
+                ]);
+
+                if (null === $gitRemote) {
+                    $gitRemote = new GitRemote($remote);
+                    try {
+                        $this->entityManager->persist($gitRemote);
+                        $gitRemote->setUrl($remote);
+                    } catch (\Doctrine\DBAL\Exception\UniqueConstraintViolationException $e) {
+                        var_dump($e);
```
Comment (turegjorup) [inline src/Handler/GitHandler.php:62]: This is debug code, should be removed or converted to proper exception handling.

File: src/Handler/GitHandler.php [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+                        continue;
+                    } catch (\Exception $e) {
+                        var_dump($e);
```
Comment (turegjorup) [inline src/Handler/GitHandler.php:65]: This is debug code, should be removed or converted to proper exception handling.

File: src/Entity/ModuleVersion.php [hunk 11]
```diff
@@ -21,6 +22,25 @@ class ModuleVersion extends AbstractBaseEntity
...
 
+    public function __toString(): string
+    {
+        return $this->getVersion();
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function display(int $style): string
```
Comment (turegjorup) [inline src/Entity/ModuleVersion.php:33]: Using integers doesn't relay what the parameter does. Using a value like `detail` does that better. Also please document the parameter with a proper phpDoc block https://docs.phpdoc.org/3.0/guide/getting-started/what-is-a-docblock.html

File: src/Twig/AppExtension.php [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+    public function getFunctions(): array
+    {
+        return [
+            new TwigFunction('admin_detail_url', [$this, 'url']),
+            new TwigFunction('entity_display', [$this, 'entity_display']),
+            new TwigFunction('has_display', [$this, 'has_display']),
+        ];
+    }
+
+    public function has_display(AbstractBaseEntity $entity): bool
```
Comment (turegjorup) [inline src/Twig/AppExtension.php]: Should be `hasDisplay` to follow standard naming

File: src/Twig/AppExtension.php [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+    {
+        return method_exists($entity, 'display');
+    }
+
+    public function entity_display(AbstractBaseEntity $entity, string $action): string
```
Comment (turegjorup) [inline src/Twig/AppExtension.php]: Should be `entityDisplay` to follow standard naming

**Review by turegjorup — CHANGES_REQUESTED** [review]

File: composer.json (hunk 8 above)
Comment (afrje) [reply composer.json:105]: Was your suggesting because "coding-standards-apply" execute the same command.

**Review by turegjorup — APPROVED** [review]

**Pull request merged**